pkg/bigquery/datasource.go

```diff
@@ -84,6 +87,13 @@ func (s *BigQueryDatasource) Connect(config backend.DataSourceInstanceSettings,
 
 	connectionKey := fmt.Sprintf("%d/%s:%s", config.ID, connectionSettings.Location, connectionSettings.Project)
 
+	if s.resourceManagerService == nil {
+		err := createResourceManagerService(settings, s)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I think the problem I faced when changing the data source is caused by this. The resource manager needs to be cached and retrieved based on `config.ID`.